drivers/bluetooth/hci/h4.c

```diff
@@ -32,8 +32,22 @@ LOG_MODULE_REGISTER(bt_driver);
 
 #define DT_DRV_COMPAT zephyr_bt_hci_uart
 
-/* We only care about `num_reports` and `evt_type`. */
-#define EXT_ADV_HDR_SIZE offsetof(struct bt_hci_evt_le_ext_advertising_report, adv_info[0].addr)
+struct h4_hdr {
+	struct {
+		struct bt_hci_evt_hdr evt;
+		struct bt_hci_evt_le_meta_event meta_evt;
+
+		/**
+		 * We only care about the `num_reports` and `evt_type` fields
+		 * of extended advertising event.
+		 */
+		uint8_t le_ext_adv_hdr[offsetof(struct bt_hci_evt_le_ext_advertising_report,
+				       adv_info[0].addr)];
+	} __packed;
+
+	struct bt_hci_acl_hdr acl;
+	struct bt_hci_iso_hdr iso;
+} __packed;
```

Comment:
Why didn't you keep evt/acl/iso in a union? We never process more than one packet at a time, so there shouldn't be any need to keep them in separate places in memory. At the very least something like this should have been properly described and justified in the commit message.

Reply:
Sorry, my brain is short-circuited. .

Reply:
Must be `union` not `struct`.